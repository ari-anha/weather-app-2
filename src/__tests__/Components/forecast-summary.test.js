import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../Components/forecast-summary";

describe("ForecastSummary", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={1525046400000}
        temperature="mockTemperature"
        description="mockDescription"
        icon="800"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct props", () => {
    const { getByText } = render(
      <ForecastSummary
        date={1525046400000}
        temperature="mockTemperature"
        description="mockDescription"
        icon="800"
      />
    );

    expect(getByText("Mon 30th Apr")).toHaveClass("date");
    expect(getByText("mockTemperatureÂ°c")).toHaveClass("temperature");
    expect(getByText("mockDescription")).toHaveClass("description");
    expect(getByText("???")).toHaveClass("wi wi-day-sunny icon???");
  });
});
