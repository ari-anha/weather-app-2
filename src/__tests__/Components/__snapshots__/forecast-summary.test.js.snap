// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="summary"
  >
    <span
      class="date"
    >
      mockDate
    </span>
    <span
      class="temperature"
    >
      mockTemperature
    </span>
    <span
      class="description"
    >
      mockDescription
    </span>
    <span
      class="icon"
    >
      mockIcon
    </span>
  </div>
</DocumentFragment>
`;
